File: Benchmarking/src/gen_11_21.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  __attribute__((noinline))
  double f0(const s0 &v0, const s0 &v1, const double &v2) {
    const std::vector<std::vector<double>> v4 = v0.p1;
    const std::vector<double> v14 = v4[0];
    std::vector<double> v6 = v14;
    std::vector<double> v20 = v6;
    const double v23 = v6[0];
    const std::vector<double> v18 = v4[0];
    v20[0] = v23;
    std::vector<double> v29 = v18;
    const double v22 = v20[0];
    const double v30 = v29[0];
    v20[0] = v30;
    return v22;
  }
  int main() {
    s0 v0({ { { 0.0 } }, { { 1.0 } } });
    s0 v1({ { { 2.0 } }, { { 3.0 } } });
    double v2(4.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen_11_21.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  @inline(never)
  func f0(_ v0: s0, _ v1: s0, _ v2: Double) -> Double {
    let v4: [[Double]] = v0.p1
    let v14: [Double] = v4[0]
    var v6: [Double] = v14
    var v20: [Double] = v6
    let v23: Double = v6[0]
    let v18: [Double] = v4[0]
    v20[0] = v23
    var v29: [Double] = v18
    let v22: Double = v20[0]
    let v30: Double = v29[0]
    v20[0] = v30
    return v22
  }
  func benchmark() {
    let v0: s0 = s0(p0: [[0.0]], p1: [[1.0]])
    let v1: s0 = s0(p0: [[2.0]], p1: [[3.0]])
    let v2: Double = 4.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 4.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
